pkg/test/expectations/expectations.go

```diff
@@ -27,3 +28,18 @@ import (
 func ExpectUnavailable(env *test.Environment, instanceType string, zone string, capacityType string) {
 	Expect(env.UnavailableOfferingsCache.IsUnavailable(instanceType, fmt.Sprintf("%s-%s", fake.Region, zone), capacityType)).To(BeTrue())
 }
+
+func ExpectKubeletFlags(env *test.Environment, expectedFlags map[string]string) {
+	Expect(env.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Len()).To(Equal(1))
+	vm := env.VirtualMachinesAPI.VirtualMachineCreateOrUpdateBehavior.CalledWithInput.Pop().VM
+	customData := *vm.Properties.OSProfile.CustomData
+	Expect(customData).ToNot(BeNil())
+
+	decodedBytes, err := base64.StdEncoding.DecodeString(customData)
+	Expect(err).To(Succeed())
+	decodedString := string(decodedBytes[:])
+
+	for key, value := range expectedFlags {
+		Expect(decodedString).To(ContainSubstring(fmt.Sprintf("--%s=%s", key, value)))
+	}
+}
```

Comment:
Alternatively, we could write this to be more generic for checking any substring matches or doesnt match.  What I like about testing like this however is we can validate the flags themselves aren't malformed easily so I chose to do things this way.